### PR TITLE
depends macOS: point --sysroot to SDK

### PR DESCRIPTION
**Pull request by rex4539**
Fixes https://github.com/zcash/zcash/issues/4149
Fixes https://github.com/zcash/zcash/issues/4150
Fixes https://github.com/zcash/zcash/issues/4151
Fixes https://github.com/zcash/zcash/issues/4152

### CODE REVIEW

**Comment by Eirik0** [comment]
@zkbot try

**Comment by daira** [comment]
Note that this self-evidently can't break any platform other than macOS.

File: depends/builders/darwin.mk [hunk 1]
```diff
@@ -10,8 +10,8 @@ build_darwin_SHA256SUM = shasum -a 256
 build_darwin_DOWNLOAD = curl --location --fail --connect-timeout $(DOWNLOAD_CONNECT_TIMEOUT) --retry $(DOWNLOAD_RETRIES) -o
 
 #darwin host on darwin builder. overrides darwin host preferences.
-darwin_CC=$(shell xcrun -f clang) -mmacosx-version-min=$(OSX_MIN_VERSION)
-darwin_CXX:=$(shell xcrun -f clang++) -mmacosx-version-min=$(OSX_MIN_VERSION) -stdlib=libc++
+darwin_CC=$(shell xcrun -f clang) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(shell xcrun --show-sdk-path)
+darwin_CXX:=$(shell xcrun -f clang++) -mmacosx-version-min=$(OSX_MIN_VERSION) -stdlib=libc++ --sysroot $(shell xcrun --show-sdk-path)
```
[the remark this answers is not in the document]
Comment (rex4539) [reply depends/builders/darwin.mk]: I just tested renaming the SDK folder by adding spaces to it and the OS informed me that it couldn't find the Command Line Tools and gave me the option to reinstall them on the spot. So it looks like Apple wants the SDK to be located in a particular directory and doesn't like things moved around. I guess I could make it work by adding symlinks but I think that's way off the average user's default behavior.

**Comment by Eirik0** [comment]
@zkbot r+

**Pull request merged**